jupyter_server/gateway/managers.py

```diff
@@ -43,13 +49,37 @@ def _default_kernel_manager_class(self):
     def _default_shared_context(self):
         return False  # no need to share zmq contexts
 
+    event_schema_id = os.path.join(
+        JUPYTER_SERVER_EVENTS_URI, "gateway", "mapping_kernel_manager", "v1"
+    )
+    event_logger = Instance(EventLogger).tag(config=True)
+
+    @default("event_logger")
+    def _default_event_logger(self):
+        if self.parent and hasattr(self.parent, "event_logger"):
+            return self.parent.event_logger
+        else:
+            # If parent does not have an event logger, create one.
+            logger = EventLogger()
+            schema_path = os.path.join(
+                DEFAULT_EVENTS_SCHEMA_PATH, "gateway", "mapping_kernel_manager", "v1.yaml"
+            )
+
+            logger.register_event_schema(schema_path)
+            self.log.info("Event is registered in GatewayMappingKernelManager.")
+            return logger
+
     def __init__(self, **kwargs):
         """Initialize a gateway mapping kernel manager."""
         super().__init__(**kwargs)
         self.kernels_url = url_path_join(
             GatewayClient.instance().url, GatewayClient.instance().kernels_endpoint
         )
 
+    def emit(self, data):
```

Comment:
We may want to move emission from the "Mapping" Kernel Manager to the actual Kernel Manager, but this depends on how we decide to generalize this.  I thought, based on previous discussions we decided to emit from the specific manager instance which is why we introduced [`ServerKernelManager`](https://github.com/jupyter-server/jupyter_server/blob/6ad8006426df68583814c415848225a49f77373c/jupyter_server/services/kernels/kernelmanager.py#L706).

Reply:
Initially, I started the implementation from KernelManger, however the KernelManager object is not exposed in serverapp, and therefore, I was not able to use the event logger in server extension that was registered in serverapp module.

Reply:
Agree. Is it okay if I remove in a separate PR ? Let me test the current behavior with an extension and circle back otherwise.

Reply:
Shoot—I missed this comment. Sorry @rajmusuku. 

I went ahead and moved the event emission down into the individual kernel manager. Let me know what you think.